language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/util/documentsymbol/DocumentSymbolResolver.java

```diff
@@ -151,20 +177,146 @@
-                .map(Node::toSourceCode).collect(Collectors.joining(""))
-                + " on " + serviceDeclarationNode.expressions().stream()
-                .map(Node::toSourceCode).collect(Collectors.joining(""));
-        SymbolKind symbolKind = SymbolKind.Interface;
+        Optional<TypeDescriptorNode> typeDesc = serviceDeclarationNode.typeDescriptor();
+        StringBuilder name = new StringBuilder("service");
+        if (typeDesc.isPresent()) {
+            name.append(" ").append(typeDesc.get().toSourceCode());
+        }
+        name.append(" ").append(serviceDeclarationNode.absoluteResourcePath().stream()
+                        .map(Node::toSourceCode).collect(Collectors.joining("")))
+                .append(" on ").append(serviceDeclarationNode.expressions().stream()
+                        .map(Node::toSourceCode).collect(Collectors.joining(",")));
+        SymbolKind symbolKind = SymbolKind.Object;
         Range range = DocumentSymbolUtil.generateNodeRange(serviceDeclarationNode);
         Optional<MetadataNode> metadata = serviceDeclarationNode.metadata();
         boolean isDeprecated = metadata.isPresent() &&
                 DocumentSymbolUtil.isDeprecated(metadata.get());
         List<DocumentSymbol> children = transformMembers(serviceDeclarationNode.members());
+        return Optional.ofNullable(createDocumentSymbol(name.toString(), symbolKind, null,
+                range, range, isDeprecated, children, this.context));
+    }
+
+    @Override
+    public Optional<DocumentSymbol> transform(TypeDefinitionNode typeDefinitionNode) {
+        String name = typeDefinitionNode.typeName().text();
+        Node typeDescriptor = typeDefinitionNode.typeDescriptor();
+        SymbolKind symbolKind;
+        List<DocumentSymbol> children = new ArrayList<>();
+        switch (typeDescriptor.kind()) {
+            case RECORD_TYPE_DESC:
+                symbolKind = SymbolKind.Struct;
+                break;
+            case OBJECT_TYPE_DESC:
+                symbolKind = SymbolKind.Interface;
+                children.addAll(transformMembers(((ObjectTypeDescriptorNode) typeDescriptor).members()));
+                break;
+            default:
+                symbolKind = SymbolKind.TypeParameter;
+        }
+        Range range = DocumentSymbolUtil.generateNodeRange(typeDefinitionNode);
+        Optional<MetadataNode> metadata = typeDefinitionNode.metadata();
+        boolean isDeprecated = metadata.isPresent() &&
+                DocumentSymbolUtil.isDeprecated(metadata.get());
         return Optional.ofNullable(createDocumentSymbol(name, symbolKind, null,
                 range, range, isDeprecated, children, this.context));
     }
 
+    @Override
+    public Optional<DocumentSymbol> transform(ModuleVariableDeclarationNode moduleVariableDeclarationNode) {
+        String name = moduleVariableDeclarationNode.typedBindingPattern().bindingPattern().toSourceCode();
```

Comment:
Here, we use the binding pattern as it is with the `toSourceCode` API. Shall we check whether the user experience with the various binding pattern options? Specially the list and map binding pattern
https://ballerina.io/spec/lang/master/#binding-pattern